### PR TITLE
FEL-630

### PR DESCRIPTION
**Pull request by AlbyGiaco**

### CODE REVIEW

File: src/components/Inflenser/Campaigns.js [hunk 1]
```diff
@@ -3,8 +3,10 @@ import { useEffect, useState } from 'react';
 import { useCampaignManager } from '../../hooks/useCampaignManager';
 import { useCampaign } from '../../hooks/useCampaign';
 import { useSharedState } from '../../context/store';
+const CampaignJson = require('../../abis/LensCampaign.json');
```
Comment (luduvigo) [inline src/components/Inflenser/Campaigns.js]: Can you import ABIs in a helper and call the helper from here?

File: src/components/Inflenser/PostCard.js [hunk 2]
```diff
@@ -12,6 +12,7 @@ import { useCampaignManager } from '../../hooks/useCampaignManager';
 import { useCampaign } from '../../hooks/useCampaign';
 import { useSharedState } from '../../context/store';
 import { fetchPublication } from '../../hooks/usePublication';
+import { Container } from 'react-bootstrap';
```
Comment (luduvigo) [inline src/components/Inflenser/PostCard.js]: Please follow this ordering convention:

- external dependencies
- internal dependencies

File: src/components/layout/Layout.js [hunk 3]
```diff
@@ -12,9 +12,10 @@ import Footer from '../footer/Footer';
 
 export default function Layout() {
   const [{ account, provider }] = useSharedState();
-  const { autoLoginWallet } = useWallet();
+  const { autoLoginWallet, getDefaultProvider } = useWallet();
   useEffect(() => {
     (async () => !account && window.localStorage.getItem('walletConnected') && (await autoLoginWallet()))();
```
Comment (luduvigo) [inline src/components/layout/Layout.js]: Can't you wrap this code in one call?

File: src/hooks/getPublicationURI.js [hunk 4]
```diff
@@ -12,7 +12,12 @@ export const getPublicationURI = () => {
     let URI;
     try {
       const signer = await provider?.getSigner();
-      const LensHub = new Contract('0xDb46d1Dc155634FbC732f92E853b10B288AD5a1d', LensHubAbi, signer);
+
+      const LensHub = new Contract(
```
Comment (luduvigo) [inline src/hooks/getPublicationURI.js]: Can you declare `LensHub` just once in this file?

File: src/hooks/useCampaign.js [hunk 5]
```diff
@@ -7,8 +7,8 @@ export const useCampaign = () => {
   const [{ provider }, dispatch] = useSharedState();
 
   const getAdvertiserPayouts = async (campaignAddress) => {
-    const signer = await provider.getSigner();
-    const Campaign = new Contract(campaignAddress, CampaignJson, signer);
+    const signer = await provider?.getSigner();
```
Comment (luduvigo) [inline src/hooks/useCampaign.js:10]: Can you define `signer` and `Campaign` once in this file?

File: src/hooks/useCampaignManager.js [hunk 6]
```diff
@@ -13,7 +13,11 @@ export const useCampaignManager = () => {
 
   const getCampaigns = async (userId, postId) => {
     const signer = await provider?.getSigner();
-    const CampaignManager = new Contract(addresses.CampaignManager, CampaignManagerJson, signer);
+    const CampaignManager = new Contract(
```
Comment (luduvigo) [inline src/hooks/useCampaignManager.js]: Can you define `CampaignManager` just once in this file?

**Review by luduvigo — CHANGES_REQUESTED** [review]

File: src/hooks/useCampaign.js (hunk 5 above)
Comment (AlbyGiaco) [reply src/hooks/useCampaign.js:10]: i cant use await outside of that function

File: src/hooks/useCampaignManager.js (hunk 6 above)
Comment (AlbyGiaco) [reply src/hooks/useCampaignManager.js]: i cant use await outside of that function

**Review by luduvigo — APPROVED** [review]

**Pull request merged**